CMSgov/cms-carts-seds
# add prettying and linting

# Description

Apologies for the volume of the review. The prettying and linting requires every js/ts file to match these rules.

[Ticket](https://qmacbis.atlassian.net/jira/software/c/projects/MDCT/boards/229?modal=detail&selectedIssue=MDCT-38)

Configures pre-commit linting in the main branch.
- .pre-commit-config.yaml contains the hook stages and remote references 
- .eslintrc.yaml outlines the configuration and specific rules for the linting stage (some rules are included in the extensions, though we explicitly declare the ones we want as per the ticket)
- ~.prettierrc.yaml includes rules for the prettying step~ Removed to uses defaults instead

## How to test

Get pre-commit set up, change a file, run git commit and see if the stages trigger.

## Dependencies 

Please note the steps in the README. The pre-commit configuration is ready, though won't be triggered locally unless you run 'pre-commit' install in your local clone. Once you've done that you'll see the pre-commit stage results after each git commit.

# Get it done

Lovely code

## Code authors checklist

## Reviewers Checklist (Two different people)

## Assignee
